x-pack/plugins/licensing/public/plugin.test.ts

```diff
@@ -269,31 +323,71 @@ describe('licensing plugin', () => {
       expect(updated).toBe(false);
     });
   });
-  describe('#stop', () => {
-    it('stops polling', async () => {
+
+  describe('expired banner', () => {
+    it('does not show "license expired" banner if license is not expired.', async () => {
       const sessionStorage = coreMock.createStorage();
       plugin = new LicensingPlugin(coreMock.createPluginInitializerContext(), sessionStorage);
+
       const coreSetup = coreMock.createSetup();
-      const { license$ } = await plugin.setup(coreSetup);
+      coreSetup.http.get.mockResolvedValueOnce(
+        licenseMock.create({ license: { status: 'active', type: 'gold' } })
+      );
 
-      let completed = false;
-      license$.subscribe({ complete: () => (completed = true) });
+      const { refresh } = await plugin.setup(coreSetup);
 
-      await plugin.stop();
-      expect(completed).toBe(true);
+      const coreStart = coreMock.createStart();
+      await plugin.start(coreStart);
+
+      await refresh();
+      expect(coreStart.overlays.banners.add).toHaveBeenCalledTimes(0);
     });
 
-    it('refresh does not trigger data re-fetch', async () => {
+    it('shows "license expired" banner if license is expired only once.', async () => {
       const sessionStorage = coreMock.createStorage();
       plugin = new LicensingPlugin(coreMock.createPluginInitializerContext(), sessionStorage);
+
       const coreSetup = coreMock.createSetup();
+      const activeLicense = licenseMock.create({ license: { status: 'active', type: 'gold' } });
+      const expiredLicense = licenseMock.create({ license: { status: 'expired', type: 'gold' } });
+      coreSetup.http.get
+        .mockResolvedValueOnce(activeLicense)
+        .mockResolvedValueOnce(expiredLicense)
+        .mockResolvedValueOnce(activeLicense)
+        .mockResolvedValueOnce(expiredLicense);
+
       const { refresh } = await plugin.setup(coreSetup);
 
-      await plugin.stop();
+      const coreStart = coreMock.createStart();
+      await plugin.start(coreStart);
+
+      await refresh();
+      expect(coreStart.overlays.banners.add).toHaveBeenCalledTimes(0);
+      await refresh();
+      expect(coreStart.overlays.banners.add).toHaveBeenCalledTimes(1);
+      await refresh();
+      expect(coreStart.overlays.banners.add).toHaveBeenCalledTimes(1);
+      await refresh();
+      expect(coreStart.overlays.banners.add).toHaveBeenCalledTimes(1);
```

Comment:
unfortunately, it's almost impossible to adjust the expiration date in ES for an uploaded license, so we have to settle for integration tests.